src/term/transform/simplify_lets.rs

```diff
@@ -0,0 +1,68 @@
+use std::collections::BTreeMap;
+
+use crate::term::{Book, DefId, DefNames, Definition, Name, Pattern, Rule, Tag, Term};
+
+impl Term {
+  pub fn simplify_let(
```

Comment:
Instead of converting `let`s directly to definitions, couldn't we convert them to `match` expressions which then get converted to function definitions? I don't see a reason why tuples should be treated differently from ADTs, and using the match machinery will also take care of linearizing the matched variable.
let to match conversion is currently done for ADT patterns in the simplify_matches file.